### DISCUSSION TITLE
Wifi via iwd (iwctl/iwlist/etc) without wpa_supplicant/nm

### DISCUSSION THREAD

**Discussion by consp** [KlipperScreen/KlipperScreen#1288] [General]
Would the klipperscreen devs be interested in the patch I made for the very specific branch of klipperscreen we use for the (upcoming) flashforge adventurer 5 mod? Since we do not want to use wpa_supplicant (and definitly not nm) due to resource constrains I made a patch to have -very- basic control with iwctl/iwd. Some stuff was copied from the wpa branch, some new mostly subprocess calls to iwctl/iwlist.

If there is interest I might look into making it a proper patch, otherwise if there is no interest I'll leave it for what it is.

It should work on any machine with iwctl/iwlist but it's limited in that the /var/lib/iwd dir has to be the directory for the network configuration files. I have no way of testing it elsewhere and it would probably be only useful for resource limited devices.

see https://github.com/KlipperScreen/KlipperScreen/compare/master...consp:KlipperScreen:feature/iwd_wifi

The check for wpa_supplicant is performed by checking if wpa_supplicant has a pid.

(yes klipperscreen runs on it, it takes 1/3 of the memory but it works :)

**Comment by alfrix** [comment]
#1269 will surely break your implementation, after that is merged then maybe

**Reply by consp** [reply]
I'll have a look once that is merged, since we want to keep it in alignment with upstream it will pop up eventually. Tnx for the heads up.

At first glance it doesn't look like too much of a hassle to adapt. Should be doable. I'll let you know once that is merged and the code has been adapted to it.
